File: 9_busqueda_binaria.cs
using System;
class Program{
    static int Binaria(int[] arr, int val){
        Array.Sort(arr);
        int l=0,r=arr.Length-1;
        while(l<=r){
            int m=(l+r)/2;
            if(arr[m]==val) return m;
            else if(arr[m]<val) l=m+1;
            else r=m-1;
        }
        return -1;
    }
    static void Main(){ Console.WriteLine(Binaria(new int[]{10,20,30,40,50},40)); }
}

File: 9_busqueda_binaria.js
function binaria(arr, val){
  arr.sort((a,b)=>a-b);
  let l=0, r=arr.length-1;
  while(l<=r){
    let m=Math.floor((l+r)/2);
    if(arr[m]==val) return m;
    arr[m]<val? l=m+1 : r=m-1;
  }
  return -1;
}
console.log(binaria([10,20,30,40,50], 40));